plugins/match.hpp

```diff
@@ -261,12 +266,19 @@ template <class DataFacadeT> class MapMatchingPlugin : public BasePlugin
         osrm::matching::CandidateLists candidates_lists;
         const auto &input_coords = route_parameters.coordinates;
         const auto &input_timestamps = route_parameters.timestamps;
+        const auto &input_bearings = route_parameters.bearings;
         if (input_timestamps.size() > 0 && input_coords.size() != input_timestamps.size())
         {
             json_result.values["status"] = "Number of timestamps does not match number of coordinates .";
             return 400;
         }
 
+        if (input_bearings.size() > 0 && input_coords.size() != input_bearings.size())
+        {
+            json_result.values["status"] = "Number of bearings does not match number of coordinates .";
+            return 400;
+        }
+
```

Comment:
Actually this will work. What probably won't work would be `bearing, bearing, bearing, coordinate, coordinate, coordinate`. We should actually fix this on a grammar level.